### PR TITLE
1152 cicd create android devliery

### PR DESCRIPTION
**Pull request by ztefanie**
### Short description
Added option for automated deployment of android apps with fastlane.

### Proposed changes
- I also made adjustments in app-credentials repository. Merged the keystore files into one keystore.
- android beta delivery can now be done, by triggering delivery_beta_native circleci workflow.
- "old" way of creating releases locally with .properties file should still work
- I am having problems with naming the open testing track in the play console to the same name for bayern and nuernberg, as renaming is only possible for bayern, but not to "beta" as the track is named for nuernberg, where renaming is not possible. Still trying to figure out if and how this is possible. Edit: Not possible, added a ifelse to select the correct track.

### Side effects
- Hopefully none, please check the app, if all flavor specific stuff and so on is correctly done in the latest beta on google play.

### Resolved issues

Fixes: #1152

### CODE REVIEW

File: .circleci/src/commands/install_fvm.yml [hunk 1]
```diff
@@ -4,20 +4,25 @@ steps:
...
       name: Install Flutter
       command: fvm install
+      working_directory: ~/project/frontend/
   - save_cache:
-      key: fvm-0-{{ checksum ".fvm/fvm_config.json" }}-{{ arch }}
+      key: fvm-0-{{ checksum "frontend/.fvm/fvm_config.json" }}-{{ arch }}
       paths:
         - .fvm
         - ~/fvm/
+      working_directory: ~/project/frontend/android
```
Comment (sarahsporck) [inline .circleci/src/commands/install_fvm.yml:20]: Why this working directory? Shouldn't it be just ~/project/frontend?

File: .circleci/src/jobs/build_android.yml [hunk 2]
```diff
@@ -3,33 +3,64 @@ docker:
...
+      working_directory: ~/project/frontend/android
+  - run:
+      name: Make attached_workspace dir
+      command: mkdir -p attached_workspace
+  - run:
+      name: Move aab
+      command: mv ~/project/frontend/build/app/outputs/bundle/<< parameters.buildConfig >>Release/app-<< parameters.buildConfig >>-release.aab ~/attached_workspace/<< parameters.buildConfig >>.aab
+  - run:
+      name: to be deleted
+      command: ls
```
Comment (sarahsporck) [inline .circleci/src/jobs/build_android.yml:60]: ?

File: .circleci/src/commands/install_fvm.yml (hunk 1 above)
Comment (ztefanie) [reply .circleci/src/commands/install_fvm.yml]: Thanks, adjusted it.

**Comment by f1sh1918** [comment]
@ztefanie how can i get this version via app store (closed testing)

File: frontend/android/app/build.gradle [hunk 3]
```diff
@@ -123,12 +133,19 @@ android {
...
+    def buildConfigName = determineBuildConfigName()
+    def buildConfig = createBuildConfig(buildConfigName)
+
     defaultConfig {
         minSdkVersion 21
         targetSdkVersion 33
         multiDexEnabled true
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
+        if (project.hasProperty('VERSION_CODE')) {
```
Comment (f1sh1918) [inline frontend/android/app/build.gradle:145]: why can this code be optional? Shouldn't we always have a version code?

File: frontend/android/fastlane/Fastfile [hunk 4]
```diff
@@ -13,26 +13,176 @@
...
+      task: "bundle",
+      flavor: flavor,
+      flags: " -Ptarget=lib/main.dart",
+      build_type: "Release",
+      properties: {
+        :VERSION_CODE => version_code,
+        :VERSION_NAME => version_name,
+        :BUILD_CONFIG_NAME => build_config_name,
+        :KEYSTORE_PATH => KEYSTORE_PATH,
+        :KEYSTORE_KEY_ALIAS => flavor == 'Bayern' ? KEYSTORE_KEY_ALIAS_BAYERN : KEYSTORE_KEY_ALIAS_NUERNBERG,
```
Comment (f1sh1918) [inline frontend/android/fastlane/Fastfile:93]: i think a more generic implementation would be nice, if we gonna add a third project/flavor . Same for `KEYSTORE_KEY_PASSWORD`

File: frontend/android/fastlane/Fastfile [hunk 5]
```diff
@@ -13,26 +13,176 @@
...
+    production_delivery = options[:production_delivery]
+    build_config_name = options[:build_config_name]
+
+    if [version_name, version_code, production_delivery, build_config_name].include?(nil)
+      raise "'nil' passed as parameter! Aborting..."
+    end
+
+    application_id = sh("app-toolbelt v0 build-config to-json #{build_config_name} android | jq -rj .applicationId ")
+
+    beta_track = application_id == 'de.nrw.it.giz.ehrensache.bayern.android' ? "open-testing" : "beta"
```
Comment (f1sh1918) [inline frontend/android/fastlane/Fastfile:119]: can you please a comment here that everyone can remember why this check has to be done

**Review by f1sh1918 — COMMENTED** [review]
I couldn't find a closed beta testing version for sozialpass.
Could you please trigger one

File: frontend/android/app/build.gradle (hunk 3 above)
Comment (ztefanie) [reply frontend/android/app/build.gradle:145]: Only for fastlane builds we set the version code as a project property. When you build if locally the flutterVersionCode is used. For builds with fastlane/circleci we us the version the toolbelt calcultaes and overwrite the flutterversion

File: frontend/android/fastlane/Fastfile (hunk 4 above)
Comment (ztefanie) [reply frontend/android/fastlane/Fastfile:93]: For more flavours i highly recommend to reuse the Bayern-upload key and NOT create a new one for every app.

**Comment by ztefanie** [comment]
> Very nice implementation :) Just one question. I couldn't find a closed beta testing version for sozialpass. Could you please trigger one

We do not use closed testing. I released a version to closed testing for bayern, as the "beta"-naming which is reserved for open-testing in google play, but somehow was renamed. There is no plan to release to closed testing and this PR does not introduce publishing to closed testing. If we want to add closed testing to our workflow, lets discuss and open a seperate ticket if we decide to do so.

**Comment by f1sh1918** [comment]
> > Very nice implementation :) Just one question. I couldn't find a closed beta testing version for sozialpass. Could you please trigger one
> 
> We do not use closed testing. I released a version to closed testing for bayern, as the "beta"-naming which is reserved for open-testing in google play, but somehow was renamed, is now used for closed testing for bayern. There is no plan to release to closed testing and this PR does not introduce publishing to closed testing. If we want to add closed testing to our workflow, lets discuss and open a seperate ticket if we decide to do so.

i understand. my problem is more that i currently can't test this release via play store which i think is important, so i hoped somehow to be able to access it via store. But i also would like to publish the current beta (3.2.3) first before adding a new beta version

**Comment by ztefanie** [comment]
> > > Very nice implementation :) Just one question. I couldn't find a closed beta testing version for sozialpass. Could you please trigger one
> > 
> > 
> > We do not use closed testing. I released a version to closed testing for bayern, as the "beta"-naming which is reserved for open-testing in google play, but somehow was renamed, is now used for closed testing for bayern. There is no plan to release to closed testing and this PR does not introduce publishing to closed testing. If we want to add closed testing to our workflow, lets discuss and open a seperate ticket if we decide to do so.
> 
> i understand. my problem is more that i currently can't test this release via play store which i think is important, so i hoped somehow to be able to access it via store. But i also would like to publish the current beta (3.2.3) first before adding a new beta version (to open testing lane)

You can test this release via playstore by joining the beta programm. These releses are open betas and available for everyone joining the beta programm in the playstore.

**Comment by f1sh1918** [comment]
> > > > Very nice implementation :) Just one question. I couldn't find a closed beta testing version for sozialpass. Could you please trigger one
> > > 
> > > 
> > > We do not use closed testing. I released a version to closed testing for bayern, as the "beta"-naming which is reserved for open-testing in google play, but somehow was renamed, is now used for closed testing for bayern. There is no plan to release to closed testing and this PR does not introduce publishing to closed testing. If we want to add closed testing to our workflow, lets discuss and open a seperate ticket if we decide to do so.
> > 
> > i understand. my problem is more that i currently can't test this release via play store which i think is important, so i hoped somehow to be able to access it via store. But i also would like to publish the current beta (3.2.3) first before adding a new beta version (to open testing lane)
> 
> You can test this release via playstore by joining the beta programm. These releses are open betas and available for everyone joining the beta programm in the playstore. You will get the most recent version in the beta track.

Yes and that's basically my problem that i can not test this Release over the store now. So I can basically only install the apk. That's why I hoped to get this working via internal beta testing

File: frontend/android/fastlane/Fastfile (hunk 4 above)
Comment (f1sh1918) [reply frontend/android/fastlane/Fastfile:93]: ok so if we use the `KEYSTORE_KEY_ALIAS_BAYERN` for all except of Nuernberg we should probably switch the condition
`:KEYSTORE_KEY_ALIAS => flavor == 'Nuernberg' ? KEYSTORE_KEY_ALIAS_NUERNBERG : KEYSTORE_KEY_ALIAS_BAYERN,`

**Review by f1sh1918 — APPROVED** [review]
Looks fine to me now

**Pull request merged**